priv/script.sh

```diff
@@ -2,441 +2,489 @@
-    mkdir -p "$(dirname "$config_file")"
-    ;;
-*zsh)
-    current_shell="zsh"
-    config_file="$HOME/.zshrc"
-    ;;
+*/bash | */fish | */zsh) : ;;
+'')
+  printf >&2 "Cannot discover shell, \$SHELL is unset.\n"
+  exit 1
+  ;;
 *)
-    printf "Unsupported shell: %s\n" "$SHELL"
-    exit 1
-    ;;
+  printf >&2 "Unsupported shell: %s\n" "$SHELL"
+  exit 1
+  ;;
 esac
 
-# Add OS detection
-OS="$(uname -s)"
-
 get_package_manager() {
-  if command -v apt-get >/dev/null; then
-    echo "apt"
-  elif command -v dnf >/dev/null; then
-    echo "dnf"
-  elif command -v pacman >/dev/null; then
-    echo "pacman"
-  elif command -v apk >/dev/null; then
-    echo "apk"
+  if has apt-get; then
+    echo apt
+  elif has dnf; then
+    echo dnf
+  elif has pacman; then
+    echo pacman
+  elif has apk; then
+    echo apk
   else
-    printf "Unsupported package manager. This script requires apt, dnf, pacman, or apk.\n"
+    printf >&2 "Unsupported package manager.\n"
+    printf >&2 "This script requires apt, dnf, pacman, or apk.\n"
     exit 1
   fi
 }
 
-case "${OS}" in
-  Linux*)
-    os_type=Linux
-    package_manager=$(get_package_manager)
-    ;;
-  Darwin*)
-    os_type=macOS
-    ;;
+if [ "$os_type" = macOS ]; then
+  package_manager=homebrew
+
+  if [ "$(uname -m)" = arm64 ]; then
+    homebrew_path=/opt/homebrew/bin
+  else
+    homebrew_path=/usr/local/bin
+  fi
+else
+  package_manager="$(get_package_manager)"
+fi
+
+already_installed() {
+  case "$1" in
+  Elixir) mise_has_version elixir "$elixir_version" ;;
+  Erlang) mise_has_version erl "$erlang_version" ;;
+  git) has git ;;
+  Homebrew) has brew ;;
+  mise) has mise ;;
+  Phoenix) has mix && mix phx.new --version >/dev/null 2>&1 ;;
+  PostgreSQL) mise_has initdb ;;
+  "Xcode Command Line Tools") has xcode-select ;;
   *)
-    printf "Unsupported OS: %s\n" "${OS}"
+    printf >&2 "Unknown tool provided for install check: %s\n" "$1"
     exit 1
     ;;
-esac
+  esac
+}
 
-already_installed() {
-    case "$1" in
-    "Elixir")
-        mise which elixir >/dev/null 2>&1
-        ;;
-    "Erlang")
-        mise which erl >/dev/null 2>&1
-        ;;
-    "git")
-        which git >/dev/null 2>&1
-        ;;
-    "Homebrew")
-        which brew >/dev/null 2>&1
-        ;;
-    "mise")
-        which mise >/dev/null 2>&1
-        ;;
-    "Phoenix")
-        mix phx.new --version >/dev/null 2>&1
-        ;;
-    "PostgreSQL")
-        mise which initdb >/dev/null 2>&1
-        ;;
-    "Xcode Command Line Tools")
-        which xcode-select >/dev/null 2>&1
-        ;;
-    *)
-        printf "Invalid name argument on checking: %s\n" "$1"
-        exit 1
-        ;;
+install_deps() {
+  case "$1" in
+  Elixir)
+    case "$package_manager" in
+    apt)
+      sudo apt-get update
+      sudo apt-get install -y unzip
+      ;;
+    dnf) sudo dnf install -y unzip ;;
+    pacman) sudo pacman -Sy --noconfirm unzip ;;
+    apk) sudo apk add --no-cache unzip ;;
+    esac
```

Comment:
If `set -e` requires case "completeness" to prevent an error halt, then add `*) : ;;` to the end of the `case` block and it will swallow unknown values.